### PR TITLE
ref(timeseries): Consolidate timeseries code and make module naming consistent

### PR DESCRIPTION
**Pull request by tkaemming**
Following up on https://github.com/getsentry/snuba/pull/912#discussion_r418324329.

### CODE REVIEW

File: snuba/query/timeseries_extension.py [hunk 1]
```diff
@@ -1,19 +1,59 @@
...
 from typing import Any, Mapping, Tuple
 
-from snuba import state
-from snuba.util import parse_datetime
+from snuba import environment, state
 from snuba.query.conditions import (
-    binary_condition,
     BooleanFunctions,
     ConditionFunctions,
+    binary_condition,
```
Comment (fpacifici) [inline snuba/query/timeseries_extension.py:8]: do we usually make distinction between functions and classes when sorting imports? or are we case sensitive on sorting ?

**Review by fpacifici — APPROVED** [review]

File: snuba/datasets/discover.py [hunk 2]
```diff
@@ -29,7 +29,7 @@
 from snuba.query.processors.impact_processor import ImpactProcessor
 from snuba.query.processors.timeseries_column_processor import TimeSeriesColumnProcessor
 from snuba.query.project_extension import ProjectExtension, ProjectWithGroupsProcessor
-from snuba.query.timeseries import TimeSeriesExtension
+from snuba.query.timeseries_extension import TimeSeriesExtension
```
Comment (tkaemming) [inline snuba/datasets/discover.py:32]: This should probably be `snuba.query.extensions.{organization,project,timeseries}` but I didn't really feel like moving all of them right now. I can do that though if we'd prefer.

File: snuba/query/timeseries_extension.py (hunk 1 above)
Comment (tkaemming) [reply snuba/query/timeseries_extension.py:8]: I've just been using the vscode "Sort Imports" macro which is case sensitive (at least by default.) I personally don't have a strong opinion either way, happy to do whatever makes sense and is easiest as long as it's consistent.

Comment (fpacifici) [reply snuba/query/timeseries_extension.py:8]: I am ok with either as long as we all know the conventions we use.

**Pull request merged**